lib/helpers/formDataToStream.js

```diff
@@ -4,108 +4,210 @@ import utils from "../utils.js";
     this.headers = textEncoder.encode(headers + CRLF);
+    console.log('FormDataPart: Headers encoded, byte length:', this.headers.byteLength);
 
+    // Calculate content length
     this.contentLength = isStringValue ? value.byteLength : value.size;
+    console.log('FormDataPart: Content length:', this.contentLength);
 
+    // Total size = headers + content + trailing CRLF
     this.size = this.headers.byteLength + this.contentLength + CRLF_BYTES_COUNT;
+    console.log('FormDataPart: Total part size:', this.size);
 
     this.name = name;
     this.value = value;
   }
 
+  /**
+   * Async generator that yields the part's data as chunks
+   * Yields: headers, then content, then trailing CRLF
+   */
   async *encode(){
+    console.log('FormDataPart.encode: Starting to encode part:', this.name);
+    
+    // First, yield the headers
     yield this.headers;
+    console.log('FormDataPart.encode: Headers yielded');
 
     const {value} = this;
 
+    // Then yield the actual content
     if(utils.isTypedArray(value)) {
+      // For string values (already encoded as TypedArray), yield directly
+      console.log('FormDataPart.encode: Yielding TypedArray value');
       yield value;
     } else {
+      // For blobs/files, use readBlob to stream the content
+      console.log('FormDataPart.encode: Streaming blob/file value');
       yield* readBlob(value);
     }
 
+    // Finally, yield trailing CRLF
+    console.log('FormDataPart.encode: Yielding trailing CRLF');
     yield CRLF_BYTES;
   }
 
+  /**
+   * Escapes special characters in field/file names for HTTP headers
+   * Replaces CR, LF, and quotes with percent-encoded values
+   */
   static escapeName(name) {
-      return String(name).replace(/[\r\n"]/g, (match) => ({
+      const escaped = String(name).replace(/[\r\n"]/g, (match) => ({
         '\r' : '%0D',
         '\n' : '%0A',
         '"' : '%22',
       }[match]));
+      
+      if (escaped !== name) {
+        console.log('FormDataPart.escapeName: Escaped name from', name, 'to', escaped);
+      }
+      
+      return escaped;
   }
 }
 
+/**
+ * Converts FormData to a readable stream in multipart/form-data format
+ * This is used for uploading form data with files in Node.js
+ * 
+ * @param {FormData} form - The FormData instance to convert
+ * @param {Function} headersHandler - Callback to receive computed headers
+ * @param {Object} options - Options for boundary generation
+ * @returns {Readable} A readable stream of the multipart form data
+ */
 const formDataToStream = (form, headersHandler, options) => {
+  console.log('formDataToStream: Starting conversion');
+  console.log('formDataToStream: Options:', options);
+  
+  // Extract or generate boundary string
   const {
     tag = 'form-data-boundary',
     size = 25,
     boundary = tag + '-' + platform.generateString(size, BOUNDARY_ALPHABET)
   } = options || {};
+  
+  console.log('formDataToStream: Generated boundary:', boundary);
 
   if(!utils.isFormData(form)) {
```

Comment:
[CRITICAL] The function does not handle errors related to invalid form inputs.

**Code Smell** detected

Suggested change:

```suggestion
if (!utils.isFormData(form)) {
  throw new TypeError('form must be a FormData instance');
}
```